src/unix/fs.c

```diff
@@ -217,6 +217,54 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  int r;
+  int (*mkostemp_function)(char*, int);
+  char *path = (char*) req->path;
+  int path_length = strlen(path);
+
+  /* EINVAL can be returned for 2 reasons:
+       1. The template's last 6 characters were not XXXXXX
+       2. open() didn't support O_CLOEXEC
+     We want to avoid going to the fallback path in case
+     of 1., so it's manually checked before. */
+  if (path_length < 6 ||
+      strcmp(path + (path_length - 6), "XXXXXX"))
+    return -EINVAL;
+
+  *(int**)(&mkostemp_function) = dlsym(RTLD_DEFAULT, "mkostemp");
+
+  /* We don't care about errors, but we do want to clean them up.
+     If there has been no error, then dlerror() will just return
+     NULL. */
+  dlerror();
+
+  if (mkostemp_function != NULL) {
+    return mkostemp_function(path, UV__O_CLOEXEC);
+  }
+
+  r = mkstemp(path);
+
+  if (r == -1)
+    return -1;
+
+  if (req->cb != NULL)
+    uv_rwlock_rdlock(&req->loop->cloexec_lock);
+
+  if (r >= 0 && uv__cloexec(r, 1) != 0) {
+    r = uv__close(r);
+    if (r != 0 && r != -EINPROGRESS)
+      abort();
+    r = -1;
```

Comment:
Actually, I think there is a bug here, the same one as in `uv__fs_open()`:  this function should return 0 when `uv__close()` returns `-EINPROGRESS` because `-EINPROGRESS` is not an error, it's informational.

Perhaps we should simply change `uv__close()` to return 0 on EINTR or EINPROGRESS; there isn't really anything you need to do, or even can do, when you get that error.